src/chain/elders_info.rs

```diff
@@ -80,25 +80,32 @@ impl EldersInfo {
 
     /// Creates a new `EldersInfo` by merging this and the other one.
     pub fn merge(&self, other: &Self) -> Result<Self, RoutingError> {
-        let members = self.members.iter().chain(&other.members).cloned().collect();
+        let members = self
+            .members
+            .iter()
+            .chain(&other.members)
+            .map(|(pub_id, p2p_node)| (*pub_id, p2p_node.clone()))
+            .collect();
         Self::new(members, self.prefix.popped(), vec![self, other])
     }
 
     // WIP: remove me (or make less heavy)
     pub fn members(&self) -> BTreeSet<PublicId> {
-        self.members
-            .iter()
-            .map(P2pNode::public_id)
-            .copied()
-            .collect()
+        self.members.keys().copied().collect()
     }
 
-    pub fn p2p_members(&self) -> &BTreeSet<P2pNode> {
+    // WIP: remove me?
+    // WIP: rename to member_map?
+    pub fn p2p_members(&self) -> &BTreeMap<PublicId, P2pNode> {
         &self.members
     }
 
+    pub fn member_nodes(&self) -> impl Iterator<Item = &P2pNode> {
+        self.members.values()
+    }
+
     pub fn member_names(&self) -> BTreeSet<XorName> {
-        self.members.iter().map(P2pNode::name).copied().collect()
+        self.members.values().map(P2pNode::name).copied().collect()
```

Comment:
Need to fix performance issue there: (avoid creating a set all the time it is very expensive).
(Optionally also check that the count is enough before doing the filter if that also improve performance).

```
    /// Returns `true` if the proofs are from a quorum of this section.
    pub fn is_quorum(&self, proofs: &ProofSet) -> bool {
        // WIP: the call to members is probably to heavy?
        proofs
            .ids()
            .filter(|id| self.members().contains(id))
            .count()
            * QUORUM_DENOMINATOR
            > self.members.len() * QUORUM_NUMERATOR
    }
    /// Returns `true` if the proofs are from all members of this section.
    pub fn is_total_consensus(&self, proofs: &ProofSet) -> bool {
        proofs
            .ids()
            .filter(|id| self.members().contains(id))
            .count()
            == self.members.len()
    }
```

https://maidsafe.slack.com/archives/C03DZQTKU/p1573127337176200?thread_ts=1573063288.154600&cid=C03DZQTKU